pilot/pkg/serviceregistry/kube/controller/endpointcontroller.go

```diff
@@ -96,11 +96,12 @@ func updateEDS(c *Controller, epc kubeEndpointsController, ep interface{}, event
 		endpoints = epc.buildIstioEndpoints(ep, host)
 	}
 
+	c.RLock()
+	svc := c.servicesMap[host]
```

Comment:
nit: the 2 blocks below both only execute if `svc != nil`. Maybe we could do something like:

```golang
if svc != nil {
    // handling k8s service selecting workload entries
    if features.EnableK8SServiceSelectWorkloadEntries {
        fep := c.collectWorkloadInstanceEndpoints(svc)
        endpoints = append(endpoints, fep...)
    }

    // Handle external address update for NodePort services whose external
    // traffic policy is set to Local.
    if len(svc.Attributes.ClusterExternalPorts) > 0 &&
        svc.Attributes.ExternalTrafficPolicy == v1.ServiceExternalTrafficPolicyTypeLocal &&
        c.updateServiceNodePortAddresses(svc) {
        c.opts.XDSUpdater.SvcUpdate(string(c.opts.ClusterID), string(svc.Hostname), svc.Attributes.Namespace, event)
    }
}
```

Reply:
That's right. I updated it. However, I would like to keep `updateServiceNodePortAddresses` separate for readability. 
```go
if len(svc.Attributes.ClusterExternalPorts) > 0 &&
	svc.Attributes.ExternalTrafficPolicy == v1.ServiceExternalTrafficPolicyTypeLocal {

	// It is written this way for clarity. updateServiceNodePortAddresses has side-effect
	// where it can update the node port addresses of the passed service. If it is updated,
	// then "addressUpdated" would be true and we have to trigger Service Update [XDS]
	if addressesUpdated := c.updateServiceNodePortAddresses(svc); addressesUpdated {
		c.opts.XDSUpdater.SvcUpdate(shard, string(svc.Hostname), svc.Attributes.Namespace, event)
	}
}
```